seismic_hazard_analysis/site_source.py

```diff
@@ -0,0 +1,977 @@
+        Section ids of the segments
+    site_nztm_coords: array of floats
+        Coordinates of points of interest
+        Shape: [3] (x, y, 0)
+
+    Returns
+    -------
+    dataframe
+        Dataframe of site-source distances
+    """
+    # Compute segment-properties
+    segment_trace_length = (
+        np.linalg.norm(
+            segment_nztm_coords[0, :2, :] - segment_nztm_coords[2, :2, :], axis=0
+        )
+        / 1e3
+    )
+    segment_strike, segment_strike_vec = compute_segment_strike_nztm(
+        segment_nztm_coords
+    )
+
+    # Compute the segment distances
+    (
+        segment_rjb_values,
+        segment_rrup_values,
+        segment_rx_values,
+        segment_ry_values,
+        segment_ry_origins,
+    ) = compute_segment_distances(
+        segment_nztm_coords, segment_strike, segment_strike_vec, site_nztm_coords
+    )
+
+    scenario_rjb, scenario_rrup, scenario_Rx, scenario_Ry = compute_scenario_distances(
+        scenario_ids,
+        nb.typed.List(scenario_section_ids),
+        segment_nztm_coords,
+        segment_strike_vec,
+        segment_trace_length,
+        segment_section_ids,
+        segment_rjb_values,
+        segment_rrup_values,
+        segment_rx_values,
+        segment_ry_values,
+        segment_ry_origins,
+    )
+
+    df = pd.DataFrame(
+        index=scenario_ids,
+        data=np.stack([scenario_rjb, scenario_rrup, scenario_Rx, scenario_Ry], axis=1),
+        columns=["rjb", "rrup", "rx", "ry"],
+    )
+    return df
+
+@nb.njit(cache=True)
+def compute_scenario_strike(
+    trace_points: np.ndarray,
+    segment_strike_vecs: np.ndarray,
+    segment_trace_length: np.ndarray,
+    segment_section_ids: np.ndarray,
+):
+    """
+    Compute nominal strike across rupture scenario
+    Based on Spudich et al. (2015)
+    Section: Strike Discordance and Nominal Strike
+
+    Note: As this calculation potentially flips the section strike
+    vector, the segment strike vector is not modified in place
+    instead a mask of the flipped strike segments is returned
+
+    Parameters
+    ----------
+    trace_points: array of floats
+        The coordinates (NZTM) of the trace points
+        shape: [n_trace_points, 2, n_segments]
+    segment_strike_vecs
+        The strike vector of each segment
+        shape: [2, n_segments]
+    segment_trace_length
+        The length of each segment
+        shape: [n_segments]
+
+    Returns
+    -------
+    scenario_strike_vec: array of floats
+        The strike vector of the rupture scenario
+        shape: [2]
+    scenario_strike: float
+        The strike of the rupture scenario
+    scenario_origin: array of floats
+        The origin of the rupture scenario
+        shape: [2]
+    segment_strike_flip_mask: array of bools
+        A mask of the segments that have strike flipped
+        shape: [n_segments]
+    """
+    # Make matrix of all unique trace points
+    # unique_trace_points = np.unique(
```

Comment:
Is it worth still having the commented code out that requires the different solution as numba doesnt support it?

Reply:
I think so (at least in my opinion), its a) much easier to read, and b) hopefully at some stage numba will support the axis parameter.